File: app/src/main/java/fr/frogdevelopment/nihongo/dico/ui/details/SentencesAdapter.java
package fr.frogdevelopment.nihongo.dico.ui.details;

import android.content.Context;
import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.content.res.ResourcesCompat;

import java.util.List;

import fr.frogdevelopment.nihongo.dico.R;
import fr.frogdevelopment.nihongo.dico.data.rest.Sentence;

public class SentencesAdapter extends ArrayAdapter<Sentence> {

    private final LayoutInflater mInflater;
    private final Typeface kanjiFont;

    public SentencesAdapter(Context context, List<Sentence> items) {
        super(context, 0, items);

        mInflater = LayoutInflater.from(context);
        this.kanjiFont = ResourcesCompat.getFont(context, R.font.sawarabi_mincho);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        ViewHolder holder;

        if (convertView == null) {
            convertView = mInflater.inflate(android.R.layout.simple_list_item_2, parent, false);
            holder = new ViewHolder(convertView);
            holder.text1.setTypeface(kanjiFont);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }

		Sentence item = getItem(position);
        holder.text1.setText(item.japanese);
        holder.text2.setText(item.translation);

        return convertView;
    }

    private static class ViewHolder {

        private final TextView text1;
        private final TextView text2;

        private ViewHolder(View view) {
            text1 = view.findViewById(android.R.id.text1);
            text2 = view.findViewById(android.R.id.text2);
        }
    }
}

File: app/src/main/java/com/frogdevelopment/nihongo/dico/ui/details/SentencesAdapter.kt
package com.frogdevelopment.nihongo.dico.ui.details

import android.content.Context
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.res.ResourcesCompat
import androidx.recyclerview.widget.RecyclerView
import com.frogdevelopment.nihongo.dico.R
import com.frogdevelopment.nihongo.dico.data.entities.Sentence
import com.frogdevelopment.nihongo.dico.databinding.SentenceRowBinding

class SentencesAdapter(context: Context, sentences: List<Sentence>) : RecyclerView.Adapter<SentencesAdapter.ViewHolder>() {

    private val mInflater: LayoutInflater = LayoutInflater.from(context)
    private val kanjiFont: Typeface? = ResourcesCompat.getFont(context, R.font.sawarabi_mincho)
    private val mSentences: List<Sentence> = sentences

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = SentenceRowBinding.inflate(mInflater, parent, false)
        binding.japanese.typeface = kanjiFont
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.binding.sentence = mSentences[position]
    }

    override fun getItemCount(): Int {
        return mSentences.size
    }

    class ViewHolder(val binding: SentenceRowBinding) : RecyclerView.ViewHolder(binding.root)

}